config/resource_suppressions.xml

```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    
+      <!-- Suppressions inside test and it folders -->
+      <suppress checks="NewlineAtEndOfFile" files="(src[\\/]it[\\/].*rule731selfexplanatory[\\/]InputJavadocMethodCheck\.java|src[\\/]test[\\/].*Newline.*AtEndOfFile\.java|src[\\/]test[\\/].*astprinter[\\/]InputJavadocComment\.javadoc|src[\\/]test[\\/].*astprinter[\\/]InputJavadocWithError\.javadoc|src[\\/]test[\\/].*checks[\\/]InputEmptyFile\.txt|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]htmlTags[\\/].*|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]javadocTags[\\/].*|src[\\/]test[\\/].*grammars[\\/]InputRegressionEmptyAst\.txt|src[\\/]test[\\/].*grammars[\\/]javadoc[\\/]InputLeadingAsterisks\.txt|src[\\/]test[\\/].*grammars[\\/]comments[\\/]InputFullOfSinglelineComments.java|src[\\/]test[\\/].*grammars[\\/]InputEmbeddedNullChar\.java)"/>
```

Comment:
> Should I add reasons for all individual suppressions or just a general message at the top

General message if all of them have identical reasons.
`Newline.*AtEndOfFile\.java` can have its own unique reason which is because the check is dependent on them for violations.

> Imho first approach may be a good practice for future suppressions?

Explain what you mean by `first approach`.
If you mean all on one line in one big expression, then no for the reasons I stated.
If you mean in the ant XML file, then no, because it limits the type of suppressions we can do by check.